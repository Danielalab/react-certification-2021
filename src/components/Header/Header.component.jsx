import React, { useRef } from 'react';
import styled from 'styled-components';
import avatar from '../../assets/images/undraw_female_avatar.svg';
import logoDesktop from '../../assets/images/logo-desktop.png';
import logoMobile from '../../assets/images/mobile-icon.png';
// import loupeIcon from '../../assets/images/loupe.svg';
import searchIconWhite from '../../assets/images/search_white.svg';
import ToggleSwitch from '../ToggleSwitch';

const HeaderStyled = styled.header`
  background-color: #ffffff;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0.75rem 2rem;
  position: fixed;
  top: 0;
  width: 100%;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
  .margin-x {
    margin: 0 0.75rem;
  }
  .menu-hamburger {
    display: none;
  }
  @media (max-width: 576px) {
    padding: 1rem 0.75rem;

    .hide-mobile {
      display: none;
    }

    .menu-hamburger {
      display: block;
    }
  }
`;

const Form = styled.form`
  display: inline-flex;
`;

const InputSearch = styled.input`
  outline: none;
  width: 100%;
  padding: 12px 20px;
  margin: 0;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px 0 0 4px;
  box-sizing: border-box;
  &:focus {
    border-color: #928a97;
  }
  @media (max-width: 576px) {
    padding: 10px;
  }
`;

const ButtonSearch = styled.button`
  background-color: #f85f73;
  border-radius: 0 4px 4px 0;
  border: none;
  cursor: pointer;
  padding: 0.75rem;
`;

const Avatar = styled.img`
  height: 50px;
  width: 50px;
  @media (max-width: 576px) {
    height: 40px;
    width: 40px;
  }
`;

const Logo = styled.img`
  height: 35px;
`;

const Wrapper = styled.div`
  display: flex;
  align-items: center;
`;

const Header = ({ handleInputChange }) => {
  const inputEl = useRef(null);
  return (
    <HeaderStyled>
      <Wrapper>
        <picture>
          <source srcSet={logoMobile} media="(max-width: 576px)" />
          <Logo src={logoDesktop} alt="Pied Piper logo" className="margin-x" />
        </picture>
        <Form
          onSubmit={(event) => {
            event.preventDefault();
            handleInputChange(inputEl.current.value);
          }}
          className="margin-x"
        >
          <InputSearch type="text" placeholder="Search" ref={inputEl} />
          <ButtonSearch aria-label="search" type="submit">
            <img src={searchIconWhite} alt="loupe icon" />
          </ButtonSearch>
        </Form>
      </Wrapper>
      <Wrapper className="nav hide-mobile">
        <div className="margin-x hide-mobile">
          <ToggleSwitch />
        </div>
        <figure className="margin-x">
          <Avatar src={avatar} alt="current user avatar" />
        </figure>
      </Wrapper>
    </HeaderStyled>
  );
};

export default Header;
