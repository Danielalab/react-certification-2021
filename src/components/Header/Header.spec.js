import Header from './Header.component';

const { render, screen, fireEvent } = require('@testing-library/react');

describe('Header', () => {
  let container;
  const handleInputChangeMock = jest.fn();

  beforeEach(() => {
    container = render(<Header handleInputChange={handleInputChangeMock} />).container;
  });

  test('Should renders Header component', () => {
    expect(container).toMatchSnapshot();
  });

  test('Should contains a logo image', () => {
    const imageElement = screen.getByAltText(/pied piper logo/i);
    expect(imageElement).toBeInTheDocument();
  });

  test('Should contains an user avatar', () => {
    const imageElement = screen.getByAltText(/current user avatar/i);
    expect(imageElement).toBeInTheDocument();
  });

  test('Should contains a toggle switch button', () => {
    const togleSwitchButton = screen.getByRole('switch');
    expect(togleSwitchButton).toBeInTheDocument();
    expect(togleSwitchButton.children.length).toBe(2);
    expect(togleSwitchButton.firstChild).toHaveAttribute('type', 'checkbox');
  });

  test('Should contains an input', () => {
    const inputElement = screen.getByPlaceholderText(/search/i);
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute('type', 'text');
  });

  test('Should contains an button to search a video', () => {
    const buttonElement = screen.getByLabelText('search');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveAttribute('type', 'submit');

    //  change input value
    fireEvent.change(screen.getByPlaceholderText(/search/i), {
      target: { value: 'test' },
    });
    fireEvent.click(buttonElement);

    expect(handleInputChangeMock).toBeCalled();
    expect(handleInputChangeMock).toBeCalledWith('test');
  });
});
