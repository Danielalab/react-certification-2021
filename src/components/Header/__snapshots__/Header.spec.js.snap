// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should renders Header component 1`] = `
<div>
  <header
    class="sc-gtsrHT kNwwym"
  >
    <div
      class="sc-jSFjdj edKxXR"
    >
      <picture>
        <source
          media="(max-width: 576px)"
          srcset="mobile-icon.png"
        />
        <img
          alt="Pied Piper logo"
          class="sc-eCApnc elGaee margin-x"
          src="logo-desktop.png"
        />
      </picture>
      <form
        class="margin-x"
      >
        <input
          class="sc-dlnjwi ikkser"
          placeholder="Search"
          type="text"
        />
      </form>
    </div>
    <div
      class="sc-jSFjdj edKxXR nav hide-mobile"
    >
      <div
        class="margin-x hide-mobile"
      >
        <label
          class="sc-bdnxRM ibymvO switch"
          for="switch-dark-mode"
          role="switch"
        >
          <input
            id="switch-dark-mode"
            type="checkbox"
          />
          <span
            class="slider round"
          />
        </label>
      </div>
      <figure
        class="margin-x"
      >
        <img
          alt="current user avatar"
          class="sc-hKFxyN lbcAzq"
          src="undraw_female_avatar.svg"
        />
      </figure>
    </div>
  </header>
</div>
`;
